### PR TITLE
archs: MSVC 2013 and greater don't have a SSE flag

### LINKED ISSUE #311: Check Arch overrides on Windows
Check if all arch overrides are still valid for windows. E.g. AVX should be available on Appveyor, but it's ruled out.

### LINKED ISSUE #408: How to deal with missing MSVC compile flags
Clang and GCC provide compile flags for every single architecture that we care of. (SSE, SSE2, SSE3, SSSE3, ...). In contrast, MSVC only provides a few. In turn, that prevents our build system from building any kernels but the generic ones. It searches for `SSE`, doesn't find it and disables all later architectures.
[x64 MSVC arch flags](https://docs.microsoft.com/en-us/cpp/build/reference/arch-x64?view=vs-2019) are `AVX`, `AVX2`, `AVX512`.
[x86 MSVC arch flags](https://docs.microsoft.com/en-us/cpp/build/reference/arch-x86?view=vs-2019) (32bit systems) are `IA32`, `SSE`, `SSE2`, `AVX`, `AVX2`, `AVX512`. Default: `SSE2`.

Our current strategy is to select a later build flag for MSVC:
https://github.com/gnuradio/volk/blob/cb388afdc015fad8bc40a8330ecea6004f52bec3/gen/archs.xml#L116-L122
Since we can only select `AVX` or later for x64, we could just set this flag for all pre-AVX builds for MSVC. 
But what are the implications? 
Does VOLK cause `Illegal Instruction` errors on systems without AVX in that case?
Or do we have to distinguish between 32 and 64bit systems?
Can we drop 32bit support with MSVC?

### PR DESCRIPTION
**Pull request by noc0lour**
But still they support SSE built-in

### CODE REVIEW

**Comment by noc0lour** [comment]
This should resolve issue #311 (Maybe someone wants to take a look at the implementation of the failing kernel and add a commit)

File: gen/archs.xml [hunk 1]
```diff
@@ -96,7 +96,6 @@
   </check>
   <flag compiler="gnu">-mmmx</flag>
   <flag compiler="clang">-mmmx</flag>
-  <flag compiler="msvc">/arch:SSE</flag>
```
Comment (noc0lour) [inline gen/archs.xml:69]: MSVC fully removed MMX support from their compilers therefore there is no reason to check for that

**Comment by noc0lour** [comment]
I wish our tests would print the full input & full output vectors for the tests ...

**Comment by jdemel** [comment]
That seems to do the trick. At least for our CI. There's just one issue:

`qa_volk_32fc_x2_dot_prod_32fc` fails consistently.
appveyor:

> 98: Test command: C:\projects\volk\lib\volk_32fc_x2_dot_prod_32fc_test.bat "C:/projects/volk/lib/Release"
> 98: Test timeout computed to be: 10000000
> 98: RUN_VOLK_TESTS: volk_32fc_x2_dot_prod_32fc(131071,1)
> 98: generic completed in 0.2588 ms
> 98: u_sse3 completed in 0.3142 ms
> 98: u_sse4_1 completed in 0.2524 ms
> 98: u_avx completed in 0.1915 ms
> 98: u_avx_fma completed in 0.2473 ms
> 98: a_generic completed in 0.5051 ms
> 98: a_sse3 completed in 0.1908 ms
> 98: a_sse4_1 completed in 0.289 ms
> 98: a_avx completed in 0.1965 ms
> 98: a_avx_fma completed in 0.1783 ms
> 98: offset 0 in1: -78.0593 + 118.412j  in2: 98.8632 + 118.412j tolerance was: 0.01
> 98: volk_32fc_x2_dot_prod_32fc: fail on arch u_sse4_1
> 98: offset 0 in1: -78.0593 + 118.412j  in2: 98.8632 + 118.412j tolerance was: 0.01
> 98: volk_32fc_x2_dot_prod_32fc: fail on arch a_sse4_1
> 98: Best aligned arch: a_avx_fma
> 98: Best unaligned arch: u_avx
> 98/134 Test  #98: qa_volk_32fc_x2_dot_prod_32fc ........................***Failed    0.19 sec

Also GH Actions

> 98: Test command: D:\a\volk\volk\build\lib\volk_32fc_x2_dot_prod_32fc_test.bat "D:/a/volk/volk/build/lib/Release"
> 98: Test timeout computed to be: 10000000
> 98: RUN_VOLK_TESTS: volk_32fc_x2_dot_prod_32fc(131071,1)
> 98: generic completed in 0.2567 ms
> 98: u_sse3 completed in 0.2135 ms
> 98: u_sse4_1 completed in 0.2722 ms
> 98: u_avx completed in 0.1871 ms
> 98: u_avx_fma completed in 0.1779 ms
> 98: a_generic completed in 0.3433 ms
> 98: a_sse3 completed in 0.21 ms
> 98: a_sse4_1 completed in 0.272 ms
> 98: a_avx completed in 0.1371 ms
> 98: a_avx_fma completed in 0.1276 ms
> 98: offset 0 in1: 33.269 + 106.213j  in2: 61.752 + 106.214j tolerance was: 0.01
> 98: volk_32fc_x2_dot_prod_32fc: fail on arch u_sse4_1
> 98: offset 0 in1: 33.269 + 106.213j  in2: 61.752 + 106.214j tolerance was: 0.01
> 98: volk_32fc_x2_dot_prod_32fc: fail on arch a_sse4_1
> 98: Best aligned arch: a_avx_fma
> 98: Best unaligned arch: u_avx_fma
>  98/134 Test  #98: qa_volk_32fc_x2_dot_prod_32fc ........................***Failed    0.17 sec

Thus, I assume there's an issue with our SSE4.1 kernel. And moreover both, aligned and unaligned, versions fail.
I'd say that's technically a different issue from the one you address here but it'll trigger this issue for all users anyway.

This is the code in question together with the aligned version. I assume this kernel would also benefit from some house keeping.
https://github.com/gnuradio/volk/blob/24c7c845b3f925b38bfd5a5ce656cc47b3b5134f/kernels/volk/volk_32fc_x2_dot_prod_32fc.h#L309-L313

We could deal with the failing kernel in a separate issue and recommend a workaround in the mean time. 
At least on my current machine:

> RUN_VOLK_TESTS: volk_32fc_x2_dot_prod_32fc(131071,1987)
> generic completed in 605.68 ms
> u_sse_64 completed in 166.123 ms
> u_sse3 completed in 189.38 ms
> u_sse4_1 completed in 514.878 ms
> u_avx completed in 109.805 ms
> u_avx_fma completed in 102.288 ms
> a_generic completed in 606.892 ms
> a_sse_64 completed in 174.225 ms
> a_sse3 completed in 170.918 ms
> a_sse4_1 completed in 519.176 ms
> a_avx completed in 94.6316 ms
> a_avx_fma completed in 96.7605 ms
> Best aligned arch: a_avx
> Best unaligned arch: u_avx_fma

I really want to avoid the SSE4.1 kernel. But that's an Ubuntu system.

**Comment by noc0lour** [comment]
@jdemel so your results tell me that SSE on a 64-bit system is 4x faster than the SSE_4_1 implementation which also returns a "wrong" result? 
Edit: not 4x but almost 3x

**Comment by jdemel** [comment]
Yes. I did a quick test on a Ryzen 4750U and these are my results. SSE4.1 is a) more than 3x slower and b) yields incorrect results. We'd need more tests for that kernel. The test times indicate that we might have to expect this. But of course, this is not a reliable benchmark.
If the result is the same across a wider range of systems (and SSE3 is also faster than the SSE4.1 kernel), we can either remove this kernel or "deactivate" it until someone figures out if this kernel can provide a speed improvement with SSE4.1 intrinsics.

**Comment by jdemel** [comment]
On another machine with Intel i7-7700T
```
RUN_VOLK_TESTS: volk_32fc_x2_dot_prod_32fc(131071,1987)
generic completed in 496.857 ms
u_sse_64 completed in 142.242 ms
u_sse3 completed in 143.292 ms
u_sse4_1 completed in 217.779 ms
u_avx completed in 90.9861 ms
u_avx_fma completed in 89.496 ms
a_generic completed in 498.016 ms
a_sse_64 completed in 144.092 ms
a_sse3 completed in 142.746 ms
a_sse4_1 completed in 216.703 ms
a_avx completed in 80.2557 ms
a_avx_fma completed in 84.7137 ms
Best aligned arch: a_avx
Best unaligned arch: u_avx_fma
```

And finally with Intel Xeon Gold 6254
```
RUN_VOLK_TESTS: volk_32fc_x2_dot_prod_32fc(131071,1987)
generic completed in 225.983 ms
u_sse_64 completed in 155.379 ms
u_sse3 completed in 159.567 ms
u_sse4_1 completed in 223.449 ms
u_avx completed in 153.07 ms
u_avx_fma completed in 152.494 ms
a_generic completed in 214.124 ms
a_sse_64 completed in 157.293 ms
a_sse3 completed in 159.777 ms
a_sse4_1 completed in 221.282 ms
a_avx completed in 149.948 ms
a_avx_fma completed in 147.423 ms
Best aligned arch: a_avx_fma
Best unaligned arch: u_avx_fma
```
Intel CPUs seem to perform better with SSE4.1 but still, this kernel is slower than the SSE kernel.

**Review by jdemel — APPROVED** [review]
Thanks for taking care of this. LGTM.

**Review by michaelld — APPROVED** [review]
LGTM; thx!

**Pull request merged**